glue/utils/geometry.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+__all__ = ['polygon_line_intersections']
+
+
+def polygon_line_intersections(px, py, xval=None, yval=None):
+    """
+    Find all the segments of intersection between a polygon and an infinite
+    horizontal/vertical line.
+
+    The polygon is assumed to be closed
+
+    Parameters
+    ----------
+    px, py : `~numpy.ndarray`
+        The vertices of the polygon
+    xval : float, optional
+        The x coordinate of the line (for vertical lines). This should only be
+        specified if yval is not specified.
+    yval : float, optional
+        The y coordinate of the line (for horizontal lines). This should only be
+        specified if xval is not specified.
+
+    Returns
+    -------
+    segments : list
+        A list of segments given as tuples of coordinates along the line.
+    """
+
+    if xval is not None and yval is not None:
+        raise ValueError("Only one of xval or yval should be specified")
+    elif xval is None and yval is None:
+        raise ValueError("xval or yval should be specified")
+    if yval is not None:
+        return polygon_line_intersections(py, px, xval=yval)
+
+    px = np.asarray(px, dtype=float)
+    py = np.asarray(py, dtype=float)
+
+    # Make sure that the polygon is closed
+    if px[0] != px[-1] or py[0] != py[-1]:
+        px = np.hstack([px, px[0]])
+        py = np.hstack([py, py[0]])
+
+    # TODO: vectorize this later, but need to get the logic for corner cases
```

Comment:
This is now vectorized in this PR. The matplotlib ``points_inside_poly`` which we rely on doesn't always behave intuitively for points on the boundaries, but this is not a new problem since we were relying on this function before.